plugins/azure/fix_plugin_azure/resource/sql.py

```diff
@@ -0,0 +1,1293 @@
+    description: Optional[str] = field(default=None, metadata={'description': 'The private link service connection description.'})  # fmt: skip
+    status: Optional[str] = field(default=None, metadata={'description': 'The private link service connection status.'})  # fmt: skip
+
+
+@define(eq=False, slots=False)
+class AzureManagedInstancePecProperty:
+    kind: ClassVar[str] = "azure_managed_instance_pec_property"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("id"),
+        "private_endpoint_id": S("properties", "privateEndpoint", "id"),
+        "private_link_service_connection_state": S("properties", "privateLinkServiceConnectionState")
+        >> Bend(AzureManagedInstancePrivateLinkServiceConnectionStateProperty.mapping),
+        "provisioning_state": S("properties", "provisioningState"),
+    }
+    id: Optional[str] = field(default=None, metadata={"description": "Resource ID."})
+    private_endpoint_id: Optional[str] = field(default=None, metadata={"description": ""})
+    private_link_service_connection_state: Optional[AzureManagedInstancePrivateLinkServiceConnectionStateProperty] = field(default=None, metadata={'description': ''})  # fmt: skip
+    provisioning_state: Optional[str] = field(default=None, metadata={'description': 'State of the Private Endpoint Connection.'})  # fmt: skip
+
+
+@define(eq=False, slots=False)
+class AzureManagedInstanceExternalAdministrator:
+    kind: ClassVar[str] = "azure_managed_instance_external_administrator"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "administrator_type": S("administratorType"),
+        "azure_ad_only_authentication": S("azureADOnlyAuthentication"),
+        "login": S("login"),
+        "principal_type": S("principalType"),
+        "sid": S("sid"),
+        "tenant_id": S("tenantId"),
+    }
+    administrator_type: Optional[str] = field(default=None, metadata={'description': 'Type of the sever administrator.'})  # fmt: skip
+    azure_ad_only_authentication: Optional[bool] = field(default=None, metadata={'description': 'Azure Active Directory only Authentication enabled.'})  # fmt: skip
+    login: Optional[str] = field(default=None, metadata={"description": "Login name of the server administrator."})
+    principal_type: Optional[str] = field(default=None, metadata={'description': 'Principal Type of the sever administrator.'})  # fmt: skip
+    sid: Optional[str] = field(default=None, metadata={"description": "SID (object ID) of the server administrator."})
+    tenant_id: Optional[str] = field(default=None, metadata={"description": "Tenant ID of the administrator."})
+
+
+@define(eq=False, slots=False)
+class AzureServicePrincipal:
+    kind: ClassVar[str] = "azure_service_principal"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "client_id": S("clientId"),
+        "principal_id": S("principalId"),
```

Comment:
and here. (not sure if the related resource can be only service principal. I guess so given this name)